Sources/MongoSwiftSync/MongoDatabase.swift

```diff
@@ -358,4 +358,78 @@ public struct MongoDatabase {
         ).wait()
         return ChangeStream(wrapping: asyncStream, client: self.client)
     }
+
+    /**
+     * Runs an aggregation framework pipeline against this database for pipeline stages that do not require an
+     * underlying collection, such as `$currentOp` and `$listLocalSessions`.
+     *
+     * - Parameters:
+     *   - pipeline: an `[Document]` containing the pipeline of aggregation operations to perform
+     *   - options: Optional `AggregateOptions` to use when executing the command
+     *   - session: Optional `ClientSession` to use when executing this command
+     *
+     * - Warning:
```

Comment:
this actually isn't a problem in the sync API, so we can get rid of the warning. 

for synchronous cursors, we automatically call `kill` for you in the `deinit` method of the type: https://github.com/mongodb/mongo-swift-driver/blob/43ca721232954904f1d5ae754e8fa6fd56c6db6c/Sources/MongoSwiftSync/MongoCursor.swift#L138-L140

so users don't have to worry about it.

In the async API, users unfortunately have to call the method manually. That is required because of the non-deterministic way in which asynchronous tasks get scheduled and completed. You can write 
```
asyncFunctionA()
asyncFunctionB()
```
but `B` might actually execute/finish before `A` does.

 A cursor being properly killed and cleaned up requires that its parent `MongoClient` is still alive, since that client maintains e.g. a pool of connections that are needed to talk to the server. 
If we just fired off an asynchronous call to `kill` from the asynchronous cursor's `deinit`, there is no way for a user to guarantee that it got cleaned up properly before they close the parent client. Having an explicit method makes it so users can sync up the order these things happen in.